userProfile/privateProfile.mjs

```diff
@@ -29,29 +30,56 @@ router.use(
   })
 )
 
-router.get("/profile", async (req, res) => {
-  const user = await req.user 
-  if (!user) return respondWithError(res, 401, "Unauthorized user") 
+router.get("/profile", auth0Middleware(), async (req, res) => {
+  const user = await req.user
+  if (!user) return respondWithError(res, 401, "Unauthorized user")
+  
   const userObj = new User(user._id)
-  const userProfile = await userObj.getSelf()
-  res.set("Content-Type", "application/json; charset=utf-8") 
-
-  res.status(200).json(userProfile)
+  userObj.getSelf()
+  .then(userData => {  
+    res.status(200).json(userData);
+  })
+  .catch(error => { 
+    res.status(error.status || error.code || 500).json({
+      error: error.message || 'An error occurred while fetching the user data.',
+      status: error.status || 'Error'
+    });
+  })
 })
 
-router.get("/projects", async (req, res) => {
-  const {_id} = await req.user 
+router.get("/test/profile", auth0Middleware(),  (req, res) => { 
```

Comment:
Remove test junk.  This looks like a route you were using for testing something.